builtin/provisioners/puppet/resource_provisioner.go

```diff
@@ -259,16 +259,37 @@ func (p *provisioner) generateAutosignToken(certname string) (string, error) {
 }
 
 func (p *provisioner) installPuppetAgentOpenSource() error {
+	task := "puppet_agent::install"
+
+	connType := p.instanceState.Ephemeral.ConnInfo["type"]
+	if connType == "" {
+		connType = "ssh"
+	}
+
+	agentConnInfo := map[string]string{
+		"type": connType,
+		"host": p.instanceState.Ephemeral.ConnInfo["host"],
+		"user": p.instanceState.Ephemeral.ConnInfo["user"],
+	}
+
+	if p.OSType == "windows" {
+		if val, ok := p.instanceState.Ephemeral.ConnInfo["password"]; ok {
+			agentConnInfo["password"] = val
+		} else {
+			return fmt.Errorf("%s failed: you must specify a password on Windows", task)
+		}
+	}
```

Comment:
@mildwonkey  I am reasonably happy with how this has turned out and as a bonus I can give the user some better feedback if they are on Windows and haven't specified a password (otherwise, it looks like Bolt will error out in a potentially confusing way). 

But with that said, it really needs to be tested on Windows.